dieboljo/embedded-loop-station
# Feature/overdub

## Related Issues:
* [Overdub Mode](https://app.asana.com/0/1203221076104575/1203349316387123)

## Main Changes:
* Implemented overdub mode on a single track
* Switched to a fork of the Audio library, that provides better buffering when reading and writing at the same time.
* Switched to WAV files instead of RAW files. Not sure of the implications this might have on some effects.
* Moved the sketch config (particularly values that might change for different users) to a different file.

## Steps To Test:
1. In `src/main.cpp`, comment out line 44 and uncomment line 43.
2. Click the record button to start recording a track, then click record again to end the recording.
3. The recorded audio should immediately start to play back.
4. Verify that clicking play toggles play/pause.
5. Verify that clicking stop returns to the beginning of the track.
6. Start playing back the loop from any point, then click record and record new audio.
7. Click record again to stop recording.
8. Play the track, and verify that the new audio is recorded over the previous audio.

## TODO:
- [ ] Currently overdub mode is just a flag in the code, but this should be assigned to a button.
- [ ] It's not currently possible to change modes in the middle of recording. Should this be possible or should it just be reflected in the display that it will change on the next recording?
